File: applications/ios/DiceRoller/DiceRollRepository/RollRequest.swift
struct RollRequest {
    let diceSize: UInt32
    let numberOfRolls: UInt32

    func toProtobuf() -> ProtobufRoll_ProtobufRollRequest {
        let protobufRollRequest = ProtobufRoll_ProtobufRollRequest.with {
            $0.diceSize = self.diceSize
            $0.numberOfRolls = self.numberOfRolls
        }

        return protobufRollRequest
    }

    static func fromProtobuf(protobufRollRequest: ProtobufRoll_ProtobufRollRequest) -> RollRequest {
        return RollRequest(
            diceSize: protobufRollRequest.diceSize,
            numberOfRolls: protobufRollRequest.numberOfRolls
        )
    }
}

File: platform-wrappers/android/LibDiceRollerAdapter/src/main/kotlin/com/walterscarborough/libdiceroller/RollRequest.kt
package com.walterscarborough.libdiceroller

import protobuf_roll.RollRequest.ProtobufRollRequest

data class RollRequest(val diceSize: Int, val numberOfRolls: Int) {

    fun toProtobuf(): ProtobufRollRequest {
        return ProtobufRollRequest
            .newBuilder()
            .setDiceSize(this.diceSize)
            .setNumberOfRolls(this.numberOfRolls)
            .build()
    }

    companion object {
        fun fromProtobuf(protobufRollRequest: ProtobufRollRequest): RollRequest {
            return RollRequest(
                diceSize = protobufRollRequest.diceSize,
                numberOfRolls = protobufRollRequest.numberOfRolls,
            )
        }
    }
}
